changelog2appdata.py

```diff
@@ -17,17 +17,24 @@ def parse_changelog(changelog_path):
         version_match = version_re.match(l)
         entry_match = entry_re.match(l)
         if version_match is not None:
-            v, _, d = version_match.groups()
-            release = (v, datetime.datetime.strptime(d, '%d.%m.%y').date(), [])
+            version, prerelease, date = version_match.groups()
+            release = (version,
+                       prerelease,
+                       datetime.datetime.strptime(date, '%d.%m.%y').date(),
+                       [])
             releases.append(release)
         elif entry_match is not None:
-            release[2].append(entry_match.groups()[0])
+            release[3].append(entry_match.group(1))
 
     return releases
 
-def get_release_xml(version, date, changes):
+def get_release_xml(version, prerelease, date, changes):
     release = ET.Element("release")
-    release.set("version", version)
+    if prerelease is None:
+        ver_str = version
+    else:
+        ver_str = f"{version}~{prerelease}"
```

Comment:
Here we put `1.9.17~beta` to appdata version, the `~` symbol makes sure that this version is considered older than `1.9.17` (without suffix).

Reply:
As I know, telegram doesn't make releases like 1.9.17-beta and 1.9.17 later on as stable, they always bump the version so the `~` separator is probably not needed.

Reply:
So should we use other separator here, e.g. a whitespace?

Reply:
I would use `-` maybe? Or it cannot be used for some reason? What I meant is that there is no need to make sure the "beta" version is older than the new version, because the new version will bump number so it will always be higher (newer).

Reply:
I guess appstream allows arbitrary version format, it's just `~` gets special treatment, so other than `~`, the choice is purely cosmetic. Though, once selected, we probably shouldn't change version format in the future. So, `-` or `~`?

Reply:
I don't have any preference and I don't really know much about appstream specs so feel free to use what you think is the best.

Reply:
I'll stick with `~`, then, if you don't mind.